OneSignalSDK/onesignal/src/main/java/com/onesignal/outcomes/model/OSOutcomeEvent.java

```diff
@@ -1,35 +1,54 @@
-package com.onesignal;
+package com.onesignal.outcomes.model;
 
 import android.support.annotation.NonNull;
 import android.support.annotation.Nullable;
 
+import com.onesignal.influence.model.OSInfluenceType;
+
 import org.json.JSONArray;
 import org.json.JSONException;
 import org.json.JSONObject;
 
-public class OutcomeEvent {
+public class OSOutcomeEvent {
 
     private static final String SESSION = "session";
     private static final String NOTIFICATION_IDS = "notification_ids";
     private static final String OUTCOME_ID = "id";
     private static final String TIMESTAMP = "timestamp";
     private static final String WEIGHT = "weight";
 
-    private OSSessionManager.Session session;
+    private OSInfluenceType session;
     private JSONArray notificationIds;
     private String name;
     private long timestamp;
     private Float weight;
 
-    public OutcomeEvent(@NonNull OSSessionManager.Session session, @Nullable JSONArray notificationIds, @NonNull String name, long timestamp, float weight) {
+    public OSOutcomeEvent(@NonNull OSInfluenceType session, @Nullable JSONArray notificationIds, @NonNull String name, long timestamp, float weight) {
         this.session = session;
         this.notificationIds = notificationIds;
         this.name = name;
         this.timestamp = timestamp;
         this.weight = weight;
     }
 
-    public OSSessionManager.Session getSession() {
+    public static OSOutcomeEvent fromOutcomeEventParams(OSOutcomeEventParams outcomeEventParams) {
```

Comment:
This name `fromOutcomeEventParams` isn't clear. Did you mean something like `fromOutcomeEventParamsFilteredToNotificationSource`? This might be better to live in another spot completely too to filter down to just Notification sources.

Reply:
this is a mapping from OSOutcomeEventsParams to OutcomeEvent , should be something like fromOutcomeEventParamsV2toOutcomeEventV1

Reply:
@Jeasmine Yes, this is more clear. Could you add a comment too to the method? Something like "Creates a new `OSOutcomeEvent` with only `notification_ids` if it is influenced to make it compatible with the V1 measure endpoint."